wherehows-web/app/components/visualization/charts/score-gauge.ts

```diff
@@ -119,24 +119,44 @@ export default class VisualizationChartsScoreGauge extends Component {
    */
   chartData: Array<IHighChartsDataConfig>;
 
-  constructor() {
-    super(...arguments);
-
+  /**
+   * Performs update functions for the charts upon initial and subsequent renders
+   */
+  updateChart(): void {
     const chartOptions = getBaseGaugeConfig();
     const chartData = getBaseChartDataConfig('score');
-    const maxScore = typeof this.maxScore === 'number' ? this.maxScore : 100;
-    const score = this.score || NaN;
+    const score = this.score || 0;
+
     // Adds our information to the highcharts formatted configurations so that they can be read in the chart
-    chartOptions.yAxis.max = maxScore;
+    chartOptions.yAxis.max = get(this, 'maxScore');
     chartData[0].data = [score];
 
     setProperties(this, {
-      score,
-      maxScore,
       chartOptions,
-      chartData,
+      chartData
+    });
+  }
+
+  /**
+   * Allows us to rerender the graph when the score gets updated. Useful for when API calls haven't been
+   * resolved yet at initial render
```

Comment:
My 2 cents on the second part of this comment, is the component should not render at all, until the api calls have resolved. Having it render inaccurate information, can be seen as a bug imo. I would wrap this in a container component that does the data fetching, and while pending indicate via a loading indicator that requests are pending.
If the request does not resolve for a while, whatever is shown can be construed as valid, which is not quite a good UX...

Reply:
I agree. Using the task's `isIdle` to only show the child component that contains the graph once we're done loading